packages/api/src/activities/activity.ts

```diff
@@ -333,16 +336,42 @@ export class Activity<T extends string = string> implements IActivity<T> {
     return this;
   }
 
+  /**
+   * Get or create the base message entity.
+   */
+  private ensureSingleRootLevelMessageEntity(): MessageEntity {
```

Comment:
nit: privates should always come after public methods

Reply:
this is interesting logic, but I think its validating in the client something we should either be validating in the server (not sure that it is) or throw a hard exception. If this is invalid here it seems like the logic says it would just silently do nothing.

Reply:
So basically if i do something like
`addAIGenerated().addCitation(...).addCitation(...)`, the logic about it being a root-entity etc etc, is all abstracted away from the developer. The library is saying it's managing all that for the developer. So I think we need to make sure we only keep one of these.

The backend is already throwing a hard exception when this happens

Reply:
oh I see what you are saying now, that makes sense